src/MicroKernel.php

```diff
@@ -24,19 +28,83 @@ public function __construct()
 
         $this->initRoutes();
         $this->initServices();
+        $this->initDB();
     }
 
     private function initRoutes(): void
     {
-        $this->routes->add('route_name', new Route('/{argument}', ['_controller' => 'App\Controller\MyController']));
+        $this->routes->add('test', new Route('/test', ['_controller' => 'App\Controller\MyController']));
         // Add your Routes here. documentation here https://symfony.com/doc/4.2/components/routing.html
+
+        $this->routes->add('homepage', new Route('/', ['_controller' => 'App\Controller\HomepageController']));
+        //$this->routes->add('route_name', new Route('/', ['_controller' => 'App\Controller\MyController']));        
     }
 
     private function initServices(): void
     {
+        // Add your Services here. documentation here https://symfony.com/doc/4.2/components/dependency_injection.html
         $this->containerBuilder->register('debug', 'App\Util\Debugger');
         $this->containerBuilder->register('entity.manager', 'App\ORM\EntityManager');
-        // Add your Services here. documentation here https://symfony.com/doc/4.2/components/dependency_injection.html
+
+        $this->containerBuilder->setParameter('cart-service.entity.manager', new Reference('entity.manager'));
+        $this->containerBuilder
+            ->register('cart-service', 'App\Service\CartService')
+            ->addArgument('%cart-service.entity.manager%');
+
+        $this->containerBuilder->setParameter('menu-service.entity.manager', new Reference('entity.manager'));
```

Comment:
it would have been a better solution to directly set the new Reference as argument instead of going with an intermediate parameter.